File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task169.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Task169 {
    
    public static List<Integer> countSmaller(int[] nums) {
        if (nums == null || nums.length == 0) {
            return new ArrayList<>();
        }
        
        int n = nums.length;
        int[] result = new int[n];
        int[] indices = new int[n];
        
        for (int i = 0; i < n; i++) {
            indices[i] = i;
        }
        
        mergeSort(nums, indices, result, 0, n - 1);
        
        List<Integer> list = new ArrayList<>();
        for (int count : result) {
            list.add(count);
        }
        return list;
    }
    
    private static void mergeSort(int[] nums, int[] indices, int[] result, int left, int right) {
        if (left >= right) {
            return;
        }
        
        int mid = left + (right - left) / 2;
        mergeSort(nums, indices, result, left, mid);
        mergeSort(nums, indices, result, mid + 1, right);
        merge(nums, indices, result, left, mid, right);
    }
    
    private static void merge(int[] nums, int[] indices, int[] result, int left, int mid, int right) {
        int[] temp = new int[right - left + 1];
        int i = left, j = mid + 1, k = 0;
        int rightCount = 0;
        
        while (i <= mid && j <= right) {
            if (nums[indices[j]] < nums[indices[i]]) {
                temp[k++] = indices[j++];
                rightCount++;
            } else {
                result[indices[i]] += rightCount;
                temp[k++] = indices[i++];
            }
        }
        
        while (i <= mid) {
            result[indices[i]] += rightCount;
            temp[k++] = indices[i++];
        }
        
        while (j <= right) {
            temp[k++] = indices[j++];
        }
        
        System.arraycopy(temp, 0, indices, left, temp.length);
    }
    
    public static void main(String[] args) {
        // Test case 1
        int[] nums1 = {5, 2, 6, 1};
        System.out.println("Test 1: " + countSmaller(nums1));
        
        // Test case 2
        int[] nums2 = {-1};
        System.out.println("Test 2: " + countSmaller(nums2));
        
        // Test case 3
        int[] nums3 = {-1, -1};
        System.out.println("Test 3: " + countSmaller(nums3));
        
        // Test case 4
        int[] nums4 = {1, 2, 3, 4, 5};
        System.out.println("Test 4: " + countSmaller(nums4));
        
        // Test case 5
        int[] nums5 = {5, 4, 3, 2, 1};
        System.out.println("Test 5: " + countSmaller(nums5));
    }
}

File: dataset/Vanilla - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task169.cpp

#include <iostream>
#include <vector>
using namespace std;

struct Node {
    int val;
    int count;
    Node* left;
    Node* right;
    
    Node(int v) : val(v), count(0), left(nullptr), right(nullptr) {}
};

class Task169 {
public:
    static Node* insert(Node* node, int val, vector<int>& result, int index, int preSum) {
        if (node == nullptr) {
            node = new Node(val);
            result[index] = preSum;
        } else if (node->val > val) {
            node->count++;
            node->left = insert(node->left, val, result, index, preSum);
        } else {
            node->right = insert(node->right, val, result, index, preSum + node->count + (node->val < val ? 1 : 0));
        }
        return node;
    }
    
    static vector<int> countSmaller(vector<int>& nums) {
        if (nums.empty()) return {};
        
        vector<int> result(nums.size(), 0);
        Node* root = nullptr;
        
        for (int i = nums.size() - 1; i >= 0; i--) {
            root = insert(root, nums[i], result, i, 0);
        }
        
        return result;
    }
};

int main() {
    // Test case 1
    vector<int> nums1 = {5, 2, 6, 1};
    vector<int> res1 = Task169::countSmaller(nums1);
    cout << "Test 1: ";
    for (int n : res1) cout << n << " ";
    cout << endl;
    
    // Test case 2
    vector<int> nums2 = {-1};
    vector<int> res2 = Task169::countSmaller(nums2);
    cout << "Test 2: ";
    for (int n : res2) cout << n << " ";
    cout << endl;
    
    // Test case 3
    vector<int> nums3 = {-1, -1};
    vector<int> res3 = Task169::countSmaller(nums3);
    cout << "Test 3: ";
    for (int n : res3) cout << n << " ";
    cout << endl;
    
    // Test case 4
    vector<int> nums4 = {1, 2, 3, 4, 5};
    vector<int> res4 = Task169::countSmaller(nums4);
    cout << "Test 4: ";
    for (int n : res4) cout << n << " ";
    cout << endl;
    
    // Test case 5
    vector<int> nums5 = {5, 4, 3, 2, 1};
    vector<int> res5 = Task169::countSmaller(nums5);
    cout << "Test 5: ";
    for (int n : res5) cout << n << " ";
    cout << endl;
    
    return 0;
}
